virttest/utils_test/__init__.py

```diff
@@ -1990,8 +1989,12 @@ def download_stress(self):
         url = self.download_url
         tmp_path = data_dir.get_tmp_dir()
         logging.info('Download stress tool from %s', url)
-        download_method = getattr(self, "_%s_download" % self.download_type)
-        download_method(url, tmp_path)
+        # If it is git based download proceed, else fall back to _download_file()
+        try:
+            download_method = getattr(self, "_%s_download" % self.download_type)
+            download_method(url, tmp_path)
+        except AttributeError:
```

Comment:
Thanks for branching it. Although this would work, but it looks a bit strange to me to treat tar/xz download as an exception, for me that should be a download method parallel to `_git_download`. I would prefer keep using `download_method` without exception and keep `_tarball_download()`, that only requires xz installation use `tarball` as it's `download_type`. Or replacing `tarball` with something like `compress`, `_compress_download()` and `self.download_type = 'compress'`.

Reply:
Or `_file_download` since it might not be compressed?